test/functional/tests/volumes/test_raid_as_cache.py

```diff
@@ -0,0 +1,72 @@
+#
+# Copyright(c) 2020 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import pytest
+
+from api.cas import casadm
+from api.cas.cache_config import CacheMode
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeSet
+from storage_devices.raid import Raid, RaidConfiguration, MetadataVariant, Level
+from test_tools import fs_utils
+from test_tools.disk_utils import Filesystem
+from test_utils.size import Size, Unit
+
+mount_point = "/mnt/test"
+test_file_path = f"{mount_point}/test_file"
+test_file_tmp_path = "/tmp/filetmp"
+test_file_size = Size(500, Unit.KiloByte)
+files_number = 100
+
+
+@pytest.mark.parametrizex("cache_mode", CacheMode)
+@pytest.mark.require_disk("core", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("cache1", DiskTypeSet([DiskType.sata, DiskType.hdd]))
+@pytest.mark.require_disk("cache2", DiskTypeSet([DiskType.sata, DiskType.hdd]))
+def test_raid_as_cache(cache_mode):
+    """
+        title: Test if SW RAID1 can be a cache device.
+        description: |
+          Test if SW RAID1 can be a cache for CAS device.
+        pass_criteria:
+          - Successful creation of RAID and building CAS device with it.
+          - Files copied successfully, the md5sum match the origin one.
+    """
+    with TestRun.step("Create RAID1."):
+        raid_disk = TestRun.disks['cache1']
+        raid_disk.create_partitions([Size(2, Unit.GibiByte)])
+        raid_disk_1 = raid_disk.partitions[0]
+        raid_disk2 = TestRun.disks['cache2']
+        raid_disk2.create_partitions([Size(2, Unit.GibiByte)])
+        raid_disk_2 = raid_disk2.partitions[0]
+
+        config = RaidConfiguration(
+            level=Level.Raid1,
+            metadata=MetadataVariant.Legacy,
+            number_of_devices=2,
+            size=Size(1, Unit.GiB))
```

Comment:
I don't think size is needed here, especially that it's set to 1GiB and RAID is being built with two devices of size 2GiB